lib/stockpile/cache.rb

```diff
@@ -22,21 +22,32 @@ module Stockpile
   module Cache
     module_function
 
-    def get(db: :default, key:)
+    def get(db: :default, key:, compress: false)
       value_from_cache = Stockpile.redis(db: db) { |r| r.get(key) }
-      Oj.load(value_from_cache) if value_from_cache
+
+      return unless value_from_cache
+
+      if compress && value_from_cache
+        Oj.load(Zlib::Inflate.inflate(Base64.decode64(value_from_cache)))
```

Comment:
Are we worried about errors trying to get existing keys that were inserted without compression? Not sure what we should do, maybe it's fine to raise a Zlib error.

Reply:
The most common case here would be trying to "upgrade" an already-in-use key, which means it would keep throwing exceptions until its regenerated.

We could fix that by including a flag about compression in the cache key, but there's no obvious way to do so that couldn't potentially conflict with a separate existing key.

Honestly not sure what the better answer is here.

Reply:
I'd rather leave this to be solved outside of gem for now and keep gem's code as simple as possible. There are ways to migrate into or out of compression (rescue from errors, compress outside of gem and then turn compression on in settings etc.) but if we bake this into gem now we will have to live with it even if we don't need it now